packages/devtools_app/lib/src/screens/network/network_controller.dart

```diff
@@ -361,28 +350,50 @@ class NetworkController extends DisposableController
 
 /// Class for managing the set of all current websocket requests, and
 /// http profile requests.
-class CurrentNetworkRequests {
-  CurrentNetworkRequests({required this.onRequestDataChange});
+class CurrentNetworkRequests extends ValueNotifier<List<NetworkRequest>> {
+  CurrentNetworkRequests() : super([]);
 
-  List<NetworkRequest> get requests => _requestsById.values.toList();
+  ValueListenable<List<NetworkRequest>> get requests =>
+      this; // todo: remove this and just let callers use .value
```

Comment:
nit: todos should be formatted as `TODO(<your username, or a github issue>): blah blah` and placed above the code they refer to

Reply:
also, do we care that the `CurrentNetworkRequests` change notifier is now public? the list of requests could be modified directly outside of this class

Reply:
I'm not sure I'm too concerned about this. normal valuelisteners for Lists suffer from the same issue. Anyone could modify the value without others being notified of the changes.

Since this is internal to Devtools only I don't think we need to be too concerned about abuse of this.